test/core/libp2p/transport/tcp_integration_test.cpp

```diff
@@ -303,51 +249,41 @@ TEST(TCP, OneTransportServerHandlesManyClients) {
   constexpr static int kSize = 1500;
   size_t counter = 0;  // number of answers
 
-  auto upgrader = makeUpgrader();
   boost::asio::io_context context(1);
-  auto e = context.get_executor();
-
-  auto transport =
-      std::make_shared<TcpTransport<decltype(e)>>(e, std::move(upgrader));
-  auto listener = transport->createListener(
-      [&](auto &&conn) {
-        expectConnectionValid(conn);
-        EXPECT_FALSE(conn->isInitiator());
-
-        EXPECT_OUTCOME_TRUE(buf, conn->readSome(kSize));
-        EXPECT_OUTCOME_TRUE(written, conn->write(buf));
-        EXPECT_EQ(written, buf.size());
-        counter++;
-
-        return outcome::success();
-      },
-      logError);
+  auto upgrader = makeUpgrader();
+  auto transport = std::make_shared<TcpTransport>(context, std::move(upgrader));
+  auto listener = transport->createListener([&](auto &&rconn) {
+    auto conn = expectConnectionValid(rconn);
+    EXPECT_FALSE(conn->isInitiator());
+
+    EXPECT_OUTCOME_TRUE(buf, conn->readSome(kSize));
+    EXPECT_OUTCOME_TRUE(written, conn->write(buf));
+    EXPECT_EQ(written, buf.size());
+    counter++;
```

Comment:
The lambda is not `mutable`, the counter should not be incremented, as I understand

Reply:
it accepts counter by ref

Reply:
If lambda is not marked mutable, the reference is const